Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp

```diff
@@ -853,6 +853,79 @@ void WebAutomationSessionProxy::getComputedLabel(WebCore::PageIdentifier pageID,
     completionHandler(std::nullopt, axObject->computedLabel());
 }
 
+void getAccessibleNode(WebCore::PageIdentifier, std::optional<WebCore::FrameIdentifier>, String nodeHandle, CompletionHandler<void(std::optional<String>, Inspector::Protocol::Automation::AccessibleNode&&)>&& completionHandler)
+{
+    String errorType;
+    RefPtr axObject = getAccessibilityObjectForNode(pageID, frameID, nodeHandle, errorType);
+
+    if (!errorType.isNull()) {
+        completionHandler(errorType, std::nullopt);
+        return;
+    }
+
+    if (!axObject) {
+        completionHandler("NodeNotFound"_s, Inspector::Protocol::Automation::AccessibleNode::create());
+        return;
+    }
+
+    struct AccessibleNodeStruct {
+        std::optional<String> activeDescendantElement;
+        std::optional<bool> atomic;
+        std::optional<String> autocomplete;
+        std::optional<bool> busy;
+        std::optional<String> checked;
+        std::optional<int> colcount;
+        std::optional<int> colspan;
+        std::optional<String> controlsElements;
+        std::optional<String> current;
+        std::optional<String> describedByElements;
+        std::optional<String> description;
+        std::optional<String> detailsElements;
+        std::optional<bool> disabled;
+        std::optional<String> errorMessageElements;
+        std::optional<String> expanded;
+        std::optional<String> flowToElements;
+        std::optional<String> haspopup;
+        std::optional<bool> hidden;
+        std::optional<String> invalid;
+        std::optional<String> keyShortcuts;
+        std::optional<String> label;
+        std::optional<String> labelledByElements;
+        std::optional<int> level;
+        std::optional<String> live;
+        std::optional<bool> modal;
+        std::optional<String> orientation;
+        std::optional<String> ownsElements;
+        std::optional<String> placeholder;
+        std::optional<int> posInSet;
+        std::optional<String> pressed;
+        std::optional<bool> readonly;
+        std::optional<String> relevant;
+        std::optional<bool> required;
+        std::optional<String> role;
+        std::optional<String> roleDescription;
+        std::optional<int> rowCount;
+        std::optional<int> rowIndex;
+        std::optional<int> rowspan;
+        std::optional<String> selected;
+        std::optional<int> setsize;
+        std::optional<String> sort;
+        std::optional<double> valueMax;
+        std::optional<double> valueMin;
+        std::optional<double> valueNow;
+        std::optional<String> valueText;
+    };
+
+    AccessibleNodeStruct axNode;
+
+    axNode.label = axObject->computedLabel();
+    axNode.role = axObject->computedRoleString();
+    axNode.checked = axObject->isChecked();
```

Comment:
Could we only set the attribute if it applies to that element? For this one you could check axObject->supportsCheckedState(). For something that isn't checkable I think it'd be more correct to leave it as empty/nullopt rather than false.